src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java

```diff
@@ -69,6 +69,63 @@ public static DocstringInfo parseDocstring(String doc, List<DocstringParseError>
     return result;
   }
 
+  /**
+   * Removes leading whitespace from docstring lines after the first.
+   *
+   * Allows docstrings to be indented to the context where they appear, without including that indent in the output.
+   * Some output formats have semantic meaning for leading whitespace (such as Markdown).
```

Comment:
This is true, but it points out a design problem. It is completely unscalable to allow different formats for the content of the docstring. Producing a unified doc set and/or pre-rendering the doc in a PR review becomes a guessing game of pick the right render engine. We should insist on markdown, or html, or anything, as long as the rendering rules for it are well defined.

Reply:
That ship sailed a long time ago I think. There are several old stardoc issues about it.

For this context we can just say "let's not introduce leading white space to accept as many inputs as possible"

Reply:
Actually. The comment is probably fine.  I was just grumbling. I'm totally in favor of this CL.

But to be honest, I am going to propose that for rules to play together well - in terms of mixing them all into a unified doc set - that we pick a single markup syntax and start converging toward that.  But we don't have to make that choice at the stardoc level. It is something that should come from the overall project and push down.